diffpy/pdfmorph/pdfmorphapp.py

```diff
@@ -355,14 +355,13 @@ def main():
                 print(save_fail_message)
                 parser.custom_error(str(e))
 
-        if outfile:
-            print(header, file=outfile)
-            numpy.savetxt(outfile, numpy.transpose([chain.xobjout, chain.yobjout]))
-            outfile.close()
-
-            if not save_stdout:
-                save_message = "\n# Morph saved to " + os.path.abspath(outfile.name)
-                print(save_message)
+        print(header, file=outfile)
+        numpy.savetxt(outfile, numpy.transpose([chain.xobjout, chain.yobjout]))
+        outfile.close()
+
+        if not save_stdout:
+            save_message = "\n# Morph saved to " + os.path.abspath(outfile.name)
```

Comment:
get in the habit of using f-strings for composing strings.  I think you didn't write this, but ok to change it.  

Also let's see if we can use a context manager for all file io if it makes sense

Finally, let's use Pathlib for handling paths if possible.  

If these changes are too disruptive, don't bother, they are syntactic sugar, but good best practices so I bring them up now.

Reply:
Got it. I was a bit wary of using f-strings since they are not available for Python 2.

Reply:
> Got it. I was a bit wary of using f-strings since they are not available for Python 2.

In general we try and follow the numpy standard to only support python versions released within the last 48 months, which means we need to support 3.9, 3.10 and 3.11 atm.   Make sure you are doing your development in a 3.11 env so we fix all deprecations as they come up....

Reply:
> get in the habit of using f-strings for composing strings. I think you didn't write this, but ok to change it.
> 
> Also let's see if we can use a context manager for all file io if it makes sense
> 
> Finally, let's use Pathlib for handling paths if possible.

New commit should address these.
Try + Except still used outside the Context Manager for custom error message.